accounts/usbwallet/wallet.go

```diff
@@ -496,6 +499,29 @@ func (w *wallet) Derive(path accounts.DerivationPath, pin bool) (accounts.Accoun
 	return account, nil
 }
 
+// ConfirmAddress implements accounts.Wallet, showing the address on the device.
+func (w *wallet) ConfirmAddress(path accounts.DerivationPath) (common.Address, error) {
+	// Try to derive the actual account and update its URL if successful
+	w.stateLock.RLock() // Avoid device disappearing during derivation
+
+	if w.device == nil {
+		w.stateLock.RUnlock()
+		return common.Address{}, accounts.ErrWalletClosed
+	}
+	<-w.commsLock // Avoid concurrent hardware access
```

Comment:
Is this a standard pattern? I'd expect that there's a `sync.Mutex` here instead of using an empty channel as a lock structure